From f68d5850a68e0e3ea7ab52552ea80cb463f5bfb0 Mon Sep 17 00:00:00 2001
From: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Date: Mon, 17 Jul 2023 17:35:39 +0300
Subject: [PATCH 15/21] libxl: Run Qemu if there is at least one virtio backend
 in Dom0

We don't know exactly what virtio backend is going to be used
(either standalone emulator or Qemu based).

Probably we need to introduce backend_type config option for
the virtio device to distinguish between various backend
implementations.

Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
---
 tools/libs/light/libxl_dm.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/tools/libs/light/libxl_dm.c b/tools/libs/light/libxl_dm.c
index f19f46c296..6a734dc7e1 100644
--- a/tools/libs/light/libxl_dm.c
+++ b/tools/libs/light/libxl_dm.c
@@ -3766,6 +3766,15 @@ int libxl__need_xenpv_qemu(libxl__gc *gc, libxl_domain_config *d_config)
         goto out;
     }
 
+    for (i = 0; i < d_config->num_virtios; i++) {
+        libxl_device_virtio *virtio = &d_config->virtios[i];
+
+        if (virtio->backend_domid == LIBXL_TOOLSTACK_DOMID) {
+            ret = 1;
+            goto out;
+        }
+    }
+
     for (idx = 0;; idx++) {
         dt = device_type_tbl[idx];
         if (!dt)
-- 
2.34.1

